### PR TITLE
chore: clean up fastfile

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [] 

### Description

A little clean up of our fastfile which was getting pretty long and hard to work with:
- moves some lengthy shared params to deliver file
- moves global config to top of file and capitalizes
- moves utility lanes and sentry lanes to separate files to keep the main fastfile just related to core delivery
- combine beta android lanes into one lane
- adds some utilities
- call the set_git_properties methods that were previously unused :) 

### Follow-ups
- would be good to move the check flags and possibly the license agreement check to release-lookout if possible
- fix the license agreement check and start running again
- suspect our sentry upload lane is still a bit overcomplicated, take another look

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- clean up fastfile - Brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (clean up fastfile - Brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3faf1e9ef40c58e00ce4580218cf1b4c51a7b6d8
</p>

**Review by gkartalis — COMMENTED** [review]
exciting! Let me know when it's ready for some 👁️

**Comment by brainbicycle** [comment]
this should be ready to go!

File: fastlane/sentry_fastlane.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                            sentry_cli_path: sentry_cli_path,
+                            org_slug: org_slug,
+                            project_slug: project_slug,
+                            version: sentry_release_name,
+                            dist: dist_version,
+                            sourcemap: [bundle_path, source_map_path],
+                            rewrite: true)
+    puts "Uploaded source js and js.map for #{project_slug}"
+  rescue StandardError => e
+    message = 'Uploading the JS bundle and/or sourcemap to Sentry failed. This sometimes happens when shipping many builds to Sentry.'
```
Comment (gkartalis) [inline fastlane/sentry_fastlane.rb:64]: **question:** is there any chance of this failing for something other than the hardcoded message? 

Would changing to handle_error(e, e.message) work better in this case?

**Review by gkartalis — APPROVED** [review]

File: fastlane/sentry_fastlane.rb (hunk 1 above)
Comment (brainbicycle) [reply fastlane/sentry_fastlane.rb:64]: yeah this message is a bit of a guess, in the handle_error function below though we do output the e.message as part of the payload so it should show both if we guessed wrong

**Pull request merged**